### DISCUSSION TITLE
Chrome Store "Best Practices" ?

### DISCUSSION THREAD

**Discussion by kurtsmock** [Cookie-AutoDelete/Cookie-AutoDelete#1650] [General]
I just got the extension and I'm pretty noob level in reading javascript, I know python much better. At any rate, I really really wanted this functionality, but the chrome store indicated this when I found the extension.
![image](https://github.com/user-attachments/assets/2bd740eb-b13b-4135-97e2-f2cbc409e584)

Now, I looked through the code and I didn't see anything dangerous to privacy or security. Can anyone give me an "Amen" on that?  I don't know what 'best practices' are not being followed, Generally I don't care too much about 'best practices' per se. I just want to make sure I don't do something stupid to open up system vulnerabilities. 

Anyone who has vetted this as well and has given it their personal thumbs up should reply so people who run into the extension like I did can find this thread and see what's up with the best practice issue. I installed it. Seems to run great. I am not running automated cleaning, I think about privacy all the time on the internet so I will keep up on it as needed. Very cool though because it was SOOOOO tedious before going through the list of cookies. Now it's one click? Yes please!!!
Thanks for creating this. You're making the world a better, more private place.

**Comment by pstemari** [comment]
The problem is that it's using Manifest v2, which is the older standard for Chrome extensions. Manifest v3 changes a bunch of the mechanics for extensions and removes a lot of functionality. In particular, Manifest v3 eliminated background tasks, and requires extension authors to change them into service workers, which aren't persistent and can't use global variables.

**Comment by brevell** [comment]
It would be good to update the Manifest. I have no idea on how to actually do this? It seems like the original author hasn't touched GitHub for quite some time. Can someone else update it @pstemari?

**Reply by pstemari** [reply]
In case I wasn't totally clear, the problem is not updating the manifest. That's about a 5-minute job. The problem is changing the code to use the new apis and stop using the old deprecated architecture.

**Reply by brevell** [reply]
Ahh makes more sense. Thanks.
